atef/tests/test_report.py

```diff
@@ -0,0 +1,34 @@
+import time
+from pathlib import Path
+from typing import Callable
+
+from atef.cache import get_signal_cache
+from atef.config import ConfigurationFile, PreparedFile
+from atef.procedure import PreparedProcedureFile, ProcedureFile
+from atef.report import ActiveAtefReport, PassiveAtefReport
+
+
+def test_demo_report(all_config_path: Path, tmp_path: Path, load_config: Callable):
+    """ smoke test to check that reports can be generated """
+    cfg = load_config(all_config_path)
+    save_path = tmp_path / 'tmp.pdf'
+    save_path.touch()
+    if isinstance(cfg, ConfigurationFile):
+        prepared_file = PreparedFile.from_config(cfg)
+        doc = PassiveAtefReport(str(save_path), config=prepared_file)
+    elif isinstance(cfg, ProcedureFile):
+        prepared_file = PreparedProcedureFile.from_origin(cfg)
+        doc = ActiveAtefReport(str(save_path), config=prepared_file)
+
+    doc.create_report()
+
+    cache = get_signal_cache()
+    cache.clear()
+    assert len(cache) == 0
+    # Give signals time to be destroyed.  Hypothesis below...
+    # If run in isolation, this can cause CA context unset, presumably caused
+    # by running a callback on a destroyed signal.  Perhaps test suite is exiting
+    # before cleanup is completed?...
+    # If run in the full suite, this test takes place in the middle of the pack,
+    # providing enough time for cleanup?
```

Comment:
Yep, this is a thing for sure.

We could maybe add a fixture for pytest teardown that ensures that everything is cleaned up, like we do in the CLI?

Reply:
I tried a variety of things here, to no immediate success.  Using this fixture and removing the sleep:
``` python
def ophyd_cleanup():
    yield
    dispatcher = ophyd.cl.get_dispatcher()
    if dispatcher is not None:
        dispatcher.stop()
```
* `pytest.fixture(scope='session', autouse=True)`
  * full suite ✅ , test_report.py ❌ (segfault)
* `pytest.fixture(scope='module', autouse=True)`
  * full suite ✅ , test_report.py ❌ (segfault) 
* `pytest.fixture(scope='function', autouse=True)`
  * full suite ❌ (test_at1l0 fails, failing to find the setpoint signal?) , test_report.py ❌ (segfault) 
* `pytest.fixture(scope='function')`, including it only in the report test
  * full suite ✅ , test_report.py ❌ (segfault)

I'm confused and may look back at this later, but I'm going to leave it for now.  Maybe the fixture is wrong?..

Reply:
`scope="session"` would have been my first attempt; anything else I'd expect to fail miserably...

Maybe add in a few debug prints to ensure it's happening at the right time?

Reply:
The `scope='session'` inexplicably worked this morning.  🤷